bluemix/configuration/core_config/cf_config.go

```diff
@@ -279,11 +299,34 @@ func (c *cfConfig) Username() (name string) {
 	return
 }
 
+// IsLoggedIn will check if the user is logged in. To determine if the user is logged in both the
+// token and the refresh token will be checked
+// If token is near expiration or expired, and a refresh token is present attempt to refresh the token.
+// If token refresh was successful, check if the new UAA token is valid. If valid, user is logged in,
+// otherwise user can be considered logged out. If refresh failed, then user is considered logged out.
+// If no refresh token is present, and token is expired, then user is considered logged out.
 func (c *cfConfig) IsLoggedIn() (loggedIn bool) {
-	c.read(func() {
-		loggedIn = c.data.AccessToken != ""
-	})
-	return
+	if token, refresh := c.UAAToken(), c.UAARefreshToken(); token != "" || refresh != "" {
+		uaaTokenInfo := NewUAATokenInfo(token)
+		if uaaTokenInfo.hasExpired() && refresh != "" {
+			refreshedToken, err := c.refreshToken(token)
+			if err != nil {
+				return false
+			}
+
+			uaaToken := fmt.Sprintf("%s %s", refreshedToken.TokenType, refreshedToken.AccessToken)
+			c.SetUAAToken(uaaToken)
+			c.SetUAARefreshToken(refreshedToken.RefreshToken)
+
+			return true
```

Comment:
again we need check again whether the refreshed UAA token is expired or not.

Reply:
Will do